### ISSUE TITLE
aria-current=page attribute is returned for all links

### ISSUE THREAD

**Issue by Jarzka** [TanStack/router#1653]
### Describe the bug

I tried to upgrade TanStack router from `1.26.7` to `1.33.7` and noticed that when rendering Breadcrumb links via `router.state.matches` or getting link props with `useLinkProps`, a new attribute `aria-current="page"` is always returned, no matter if the page is actually an active page or not.

For example, I'm currently on page `/fi/123/hello-world` and I make this call

```ts
  const eventLinkProps = useLinkProps({
    to: "/$language/$event",
    params: {
      language,
      event,
    },
  });
```

I get a link props object which contains `aria-current: "page"` even if my current page is `/fi/123/hello-world`, not `/fi/123`

I guess this bug is related to this change: https://github.com/TanStack/router/releases/tag/v1.26.20

### Your Example Website or App

-

### Steps to Reproduce the Bug or Issue

-

### Expected behavior

`aria-current="page"` should be returned only when my current location is the current page. Alternatively, there should be a way to omit it completely if I want to handle aria logic by myself.

### Platform

- OS: macOS
- Browser: Chrome 124.0.6367.119

**Comment by SeanCassiere** [comment]
Tested this with version [1.33.7](https://github.com/TanStack/router/releases/tag/v1.33.7) and this feature is working as expected.

What you are finding being returned is related to your `activeOptions` on the `<Link>` component (or using the hook).

If you want to handle the aria logic yourself, you are able to `delete` the key from the returned props in the useLinkProps hook.

**Comment by Jarzka** [comment]
I have not used `activeOptions` for any of my `Link` or `useLinkProps` call so I'm using the default options. It seems that this problem can be fixed by using using `{ activeOptions: { exact: true} }` option.
